lib/decorators-serialization.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const mdsf = require('mdsf');
+
+const objFilter = (obj, blockedKeys) => {
+  if (!blockedKeys) return obj;
+  return Object.keys(obj).reduce((newObj, key) => {
+    if (!blockedKeys.includes(key)) newObj[key] = obj[key];
+    return newObj;
+  }, {});
+};
+
+const extractDefinition = ins => mdsf.stringify(objFilter(ins, ['definition']));
+
+const domains = {
+  Validate: ins => `Validate(${ins.validate.toString()})`,
+  Enum: ins => `Enum('${ins.values.join("','")}')`,
+  Flags: ins => {
+    const { enum: enumDomain, ...values } = ins;
+    return `Flags(${values},${enumDomain})`;
+  },
+};
+
+const category = {
+  Index: ins => `Index('${ins.fields.join("','")}')`,
```

Comment:
IMHO, wrapping into a `decorator` should be done elsewhere automatically, every decorator is wrapped into `${decoratorName}()`.